File: src/gestorAplicación/externo/Estado.java
package gestorAplicación.externo;

import java.io.Serializable;
import java.util.ArrayList;

public class Estado implements Serializable {
	private static final long serialVersionUID = 1L;
	public static final String nombreD = "Estados";
	private static ArrayList<Estado> estadosTotales = new ArrayList<Estado>();
	private String nombre;
	private int id;
	private double tasa_impuestos;
	private Divisas divisa;
	
	//Funcionalidad Compra Cartera
	//Atributo que almacena el interes bancario corriente determinado por cada Estado
	private double interes_bancario_corriente;
	//Atributo que almacena el multiplicador máximo aplicable al interes_bancario_corriente
	private double tasas_usura;
	
	//Constructor
	public Estado(String nombre, double tasa_impuestos, Divisas divisa) {
		this.setNombre(nombre);
		this.setTasa_impuestos(tasa_impuestos);
		this.setDivisa(divisa);
		Estado.getEstadosTotales().add(this);
		this.setId(Estado.getEstadosTotales().size());
	}
	public Estado(){}
	
	//Gets
	public String getNombre() {
		return nombre;
	}
	
	public double getTasa_impuestos() {
		return tasa_impuestos;
	}

	public Divisas getDivisa() {
		return divisa;
	}
	
	//Sets
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	
	public void setTasa_impuestos(double tasa_impuestos) {
		this.tasa_impuestos = tasa_impuestos;
	}

	public void setDivisa(Divisas divisa) {
		this.divisa = divisa;
	}

	public static ArrayList<Estado> getEstadosTotales() {
		return estadosTotales;
	}
	
	public static void setEstadosTotales(ArrayList<Estado> estadosTotales) {
		Estado.estadosTotales = estadosTotales;
	}
	
	public int getId() {
		return id;
	}
	
	public void setId(int id) {
		this.id = id;
	}
	
	public double getInteres_bancario_corriente() {
		return interes_bancario_corriente;
	}
	
	public void setInteres_bancario_corriente(double interes_bancario_corriente) {
		this.interes_bancario_corriente = interes_bancario_corriente;
	}
	
	public double getTasas_usura() {
		return tasas_usura;
	}
	
	public void setTasas_usura(double tasas_usura) {
			this.tasas_usura = tasas_usura;
		}
	
}

File: Python/src/gestorAplicación/externo/estado.py
class Estado():
    _estadosTotales = []
    def __init__(self,nombre,tasa_impuestos,divisa):
        self._nombre = nombre
        self._tasa_impuestos = tasa_impuestos
        self._divisa = divisa
        self._id = len(Estado.estadosTotales)
        Estado.estadosTotales.append(self)
        self._interes_bancario_corriente
        self._tasa_usura

    def __str__(self):
        return f"{self.getNombre} Divisa: {self.getDivisa}"


# Set y Gest
    def getNombre(self):
        return self._nombre
    
    def getTasa_impuestos(self):
        return self._tasa_impuestos
    
    def getDivisa(self):
        return self._divisa
    
    def getId(self):
        return self._id
    
    def getEstadoTotales(self):
        return Estado._estadosTotales
    
    def getTasas_usura(self):
        return self._tasa_usura
    
    def getInteres_bancario_corriente(self):
        return self._Interes_bancario_corriente

    def setInteres_bancario_corriente(self,interes_bancario_corriente):
        self._interes_bancario_corriente= interes_bancario_corriente

    def setTasa_usura(self,tasa_usura):
        self._tasa_usura= tasa_usura

    def setTasa_impuestos(self,tasa_impuestos):
        self._tasa_impuestos= tasa_impuestos
        
    def setDivisa(self,divisa):
        self._divisa= divisa
    
    def setId(self,id):
        self._id= id
        
    def setNombre(self,nombre):
        self._nombre= nombre

